public/app/features/auth-config/ProviderConfigForm.tsx

```diff
@@ -117,92 +141,121 @@ export const ProviderConfigForm = ({ config, provider, isLoading }: ProviderConf
-                    control={control}
-                    errors={errors}
-                    setValue={setValue}
-                    register={register}
-                    watch={watch}
-                    unregister={unregister}
-                    provider={provider}
-                    secretConfigured={!!config?.settings.clientSecret}
-                  />
+                  <CollapsableSection label={section.name} isOpen={index === 0} key={section.name}>
+                    {section.fields
+                      .filter((field) => (typeof field !== 'string' ? !field.hidden : true))
+                      .map((field) => {
+                        return (
+                          <FieldRenderer
+                            key={typeof field === 'string' ? field : field.name}
+                            field={field}
+                            control={control}
+                            errors={errors}
+                            setValue={setValue}
+                            register={register}
+                            watch={watch}
+                            unregister={unregister}
+                            provider={provider}
+                            secretConfigured={!!config?.settings.clientSecret}
+                          />
+                        );
+                      })}
+                  </CollapsableSection>
                 );
               })}
-            </>
-          )}
-          <Box display={'flex'} gap={2} marginTop={6}>
+          </Stack>
+        ) : (
+          <>
+            {providerFields.map((field) => {
+              return (
+                <FieldRenderer
+                  key={field}
+                  field={field}
+                  control={control}
+                  errors={errors}
+                  setValue={setValue}
+                  register={register}
+                  watch={watch}
+                  unregister={unregister}
+                  provider={provider}
+                  secretConfigured={!!config?.settings.clientSecret}
+                />
+              );
+            })}
+          </>
+        )}
+        <Box display={'flex'} gap={2} marginTop={6}>
+          {!config?.settings.enabled ? (
             <Field>
-              <Button type={'submit'} disabled={isSaving}>
-                {isSaving ? 'Saving...' : 'Save'}
+              <Button type={'submit'} onClick={() => setValue('enabled', true)} disabled={isSaving}>
+                {isSaving ? 'Saving...' : 'Save and enable'}
               </Button>
             </Field>
-            <Field>
-              <LinkButton href={'/admin/authentication'} variant={'secondary'}>
-                Discard
-              </LinkButton>
-            </Field>
+          ) : (
             <Field>
               <Button
+                type={'submit'}
+                disabled={isSaving}
+                onClick={() => setValue('enabled', false)}
                 variant={'secondary'}
```

Comment:
I think this'd be destructive variant, otherwise looks a bit weird imo
Before:
![Screenshot 2024-02-29 at 15 58 04](https://github.com/grafana/grafana/assets/8878045/aa890e25-3ed8-4ab0-b8d3-a4e9816ed93a)
After:

![Screenshot 2024-02-29 at 16 00 36](https://github.com/grafana/grafana/assets/8878045/73fc736f-e2cb-4443-88ce-3b1b809080fd)

Reply:
It is intentionally `secondary` and not a `destructive` (we have a mockup/design for it).